boot/cypress/cy_flash_pal/cy_flash_map.c

```diff
@@ -0,0 +1,494 @@
+#else
+static struct flash_area bootloader =
+{
+    .fa_id = FLASH_AREA_BOOTLOADER,
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+    .fa_off = CY_BOOTLOADER_START_ADDRESS,
+    .fa_size = CY_BOOT_BOOTLOADER_SIZE
+};
+
+static struct flash_area primary_1 =
+{
+    .fa_id = FLASH_AREA_IMAGE_PRIMARY(0),
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+    .fa_off = CY_FLASH_BASE + CY_BOOT_BOOTLOADER_SIZE,
+    .fa_size = CY_BOOT_PRIMARY_1_SIZE
+};
+
+static struct flash_area secondary_1 =
+{
+    .fa_id = FLASH_AREA_IMAGE_SECONDARY(0),
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+    .fa_off = CY_FLASH_BASE +\
+                CY_BOOT_BOOTLOADER_SIZE +\
+                CY_BOOT_PRIMARY_1_SIZE,
+    .fa_size = CY_BOOT_SECONDARY_1_SIZE
+};
+#if (MCUBOOT_IMAGE_NUMBER == 2) /* if dual-image */
+static struct flash_area primary_2 =
+{
+    .fa_id = FLASH_AREA_IMAGE_PRIMARY(1),
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+    .fa_off = CY_FLASH_BASE +\
+                CY_BOOT_BOOTLOADER_SIZE +\
+                CY_BOOT_PRIMARY_1_SIZE +\
+                CY_BOOT_SECONDARY_1_SIZE,
+    .fa_size = CY_BOOT_PRIMARY_2_SIZE
+};
+
+static struct flash_area secondary_2 =
+{
+    .fa_id = FLASH_AREA_IMAGE_SECONDARY(1),
+    /* TODO: it is for external flash memory
+    .fa_device_id = FLASH_DEVICE_EXTERNAL_FLASH(CY_BOOT_EXTERNAL_DEVICE_INDEX), */
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+    .fa_off = CY_FLASH_BASE +\
+                CY_BOOT_BOOTLOADER_SIZE +\
+                CY_BOOT_PRIMARY_1_SIZE +\
+                CY_BOOT_SECONDARY_1_SIZE +\
+                CY_BOOT_PRIMARY_2_SIZE,
+    .fa_size = CY_BOOT_SECONDARY_2_SIZE
+};
+#endif
+static struct flash_area scratch =
+{
+    .fa_id = FLASH_AREA_IMAGE_SCRATCH,
+    .fa_device_id = FLASH_DEVICE_INTERNAL_FLASH,
+#if (MCUBOOT_IMAGE_NUMBER == 1) /* if single-image */
+     .fa_off = CY_FLASH_BASE +\
+                CY_BOOT_BOOTLOADER_SIZE +\
+                CY_BOOT_PRIMARY_1_SIZE +\
+                CY_BOOT_SECONDARY_1_SIZE,
+#elif (MCUBOOT_IMAGE_NUMBER == 2) /* if dual-image */
+    .fa_off = CY_FLASH_BASE +\
+                CY_BOOT_BOOTLOADER_SIZE +\
+                CY_BOOT_PRIMARY_1_SIZE +\
+                CY_BOOT_SECONDARY_1_SIZE +\
+                CY_BOOT_PRIMARY_2_SIZE +\
+                CY_BOOT_SECONDARY_2_SIZE,
+#endif
+    .fa_size = CY_BOOT_SCRATCH_SIZE
+};
+#endif
+
+#ifdef CY_FLASH_MAP_EXT_DESC
+/* Use external Flash Map Descriptors */
+extern struct flash_area *boot_area_descs[];
+#else
+struct flash_area *boot_area_descs[] =
+{
+    &bootloader,
+    &primary_1,
+    &secondary_1,
+#if (MCUBOOT_IMAGE_NUMBER == 2) /* if dual-image */
+    &primary_2,
+    &secondary_2,
+#endif
+    &scratch,
+    NULL
+};
+#endif
+
+/*< Opens the area for use. id is one of the `fa_id`s */
+int flash_area_open(uint8_t id, const struct flash_area **fa)
+{
+    int ret = -1;
+    uint32_t i = 0;
+
+    while(NULL != boot_area_descs[i])
+    {
```

Comment:
At least the rest of the mcuboot convention places this brace at the end of the previous line.